app/src/main/java/com/hedvig/app/SplashActivity.kt

```diff
@@ -85,6 +107,34 @@ class SplashActivity : BaseActivity() {
         } ?: startDefaultActivity(loginStatus)
     }
 
+    private fun startBirthdayActivity(){
+        startActivity(Intent(this, BirthdayActivity::class.java))
+    }
+
+
+    private fun hasBenCelebrated(): Boolean {
+        val pref = applicationContext.getSharedPreferences("MyPref", 0) // 0 - for private mode
+
+        return pref.getBoolean("celebrated",false)
+    }
+
+    private fun setHasBenCelebrated() {
+        val socialSecurityNumber = "199701174617"
+
+        val currentDay = Calendar.getInstance().get(Calendar.DAY_OF_MONTH)
+        val birthDay = socialSecurityNumber.substring(6, socialSecurityNumber.length - 4).toInt()
+
+        val pref = applicationContext.getSharedPreferences("MyPref", 0) // 0 - for private mode
+        val editor: SharedPreferences.Editor = pref.edit()
+
+        if (currentDay == birthDay) {
+            editor.putBoolean("celebrated", true)
```

Comment:
This preference will make sure that the user will only be celebrated on the first birthday they open the app - we probably need to store the year that the user has been celebrated in order to allow subsequent years to receive a celebration as well 🙂